gnucash/gnome/gnc-plugin-business.c

```diff
@@ -715,6 +716,13 @@ gnc_business_assign_payment (GtkWindow *parent,
     if (xaccTransCountSplits(trans) <= 1)
         return;
 
+    if (xaccTransGetFirstAPARAcctSplit (trans, FALSE) == NULL)
+    {
+        gnc_warning_dialog (parent, "%s",
+                            _("There must be an A/P or A/R transfer account."));
+        return;
+    }
+
```

Comment:
I don't think this was ever my intention when reworking the Assign As Payment functionality (and this is even against prior advice -- avoid manually creating transactions involving the A/R or A/P accounts).

Why would you prevent such transactions from being considered for being transformed to a business payment ? From my point of view this is safe. If there's no A/R or A/P split, the user can select this in the dialog that will be shown. In that dialog it's the Post to account.

Reply:
From my testing it seemed that if there's no APAR split then the assign as payment dialog appears but clicking OK then nothing was actioned in the book.

Reply:
That's how it behaves now indeed and that's the bug IMO :)

I'll be looking into this before the 5.1 release. In my first experiment I see there's a currency conversion error showing up in the logs that shouldn't be there. That's my starting point.

Reply:
Thanks, let's consider this branch a lost cause then 🙂